web/packages/teleport/src/components/FormLogin/FormIdentifierFirst.tsx

```diff
@@ -0,0 +1,275 @@
+          onNotYou={onNotYou}
+          fetchAttempt={fetchAttempt}
+        />
+      )}
+      {connectors.length > 1 && (
+        <MultipleConnectors
+          providers={connectors}
+          onLoginWithSso={onLoginWithSso}
+          onNotYou={onNotYou}
+          fetchAttempt={fetchAttempt}
+        />
+      )}
+    </Flex>
+  );
+}
+
+function UsernamePrompt({
+  onSubmitUsername,
+  username,
+  setUsername,
+  fetchAttempt,
+}: {
+  onSubmitUsername(): void;
+  username: string;
+  setUsername: (username: string) => void;
+  fetchAttempt: Attempt<void>;
+}) {
+  const inputRef = useRefAutoFocus<HTMLInputElement>({
+    shouldFocus: true,
+  });
+
+  return (
+    <Flex
+      as="form"
+      alignItems="center"
+      justifyContent="center"
+      flexDirection="column"
+      onSubmit={e => {
+        e.preventDefault();
+        onSubmitUsername();
+      }}
+      width="100%"
+    >
+      <Text typography="h3" mb={3}>
+        Enter your username to log in with SSO
+      </Text>
+      <Input
+        ref={inputRef}
+        value={username}
+        onChange={e => setUsername(e.target.value)}
+        placeholder="Username"
+        width="80%"
+        mb={3}
+      />
+      <ButtonPrimary
+        type="submit"
+        size="medium"
+        disabled={fetchAttempt.status === 'processing' || !username.trim()}
+        width="80%"
+      >
+        Log in
+      </ButtonPrimary>
+    </Flex>
+  );
+}
+
+// OneConnector is the view for when there is a remembered user and only one connector for them.
+function OneConnector({
+  provider,
+  onLoginWithSso,
+  fetchAttempt,
+  onNotYou,
+}: {
+  provider: AuthProvider;
+  onLoginWithSso(provider: AuthProvider): void;
+  fetchAttempt: Attempt<void>;
+  onNotYou(): void;
+}) {
+  let { name, type, displayName } = provider;
+  const connectorName = displayName || name;
+  const ssoType = guessProviderType(connectorName, type);
+
+  return (
+    <Flex flexDirection="column" alignItems="center" gap={3} mt={3}>
+      <ButtonSso
+        title={`Log in with ${connectorName}`}
+        ssoType={ssoType}
+        disabled={fetchAttempt.status === 'processing'}
+        autoFocus={true}
+        onClick={e => {
+          e.preventDefault();
+          onLoginWithSso(provider);
+        }}
+      />
+      <ButtonLink
+        onClick={onNotYou}
+        disabled={fetchAttempt.status === 'processing'}
+      >
+        Not you? Click here.
+      </ButtonLink>
+    </Flex>
+  );
+}
+
+function MultipleConnectors({
+  providers,
+  onLoginWithSso,
+  onNotYou,
+  fetchAttempt,
+}: {
+  providers: AuthProvider[];
+  onLoginWithSso(provider: AuthProvider): void;
+  onNotYou(): void;
+  fetchAttempt: Attempt<void>;
+}) {
+  return (
+    <Flex flexDirection="column" gap={3}>
+      <Text textAlign="center">Select an SSO provider to continue.</Text>
+      {fetchAttempt.status === 'processing' && (
+        <Box textAlign="center" m={4}>
+          <Indicator delay="none" />
```

Comment:
Curious why we have loading indicator for `MultipleConnectors` but not for `OneConnector`